xgcm/test/test_transform.py

```diff
@@ -0,0 +1,544 @@
+        coords={case_param_dict["target_coord"][0]: case_param_dict["target_coord"][1]},
+        name=case_param_dict["target_data"][0],
+    )
+
+    # parse the 'target_data' from the actual input
+    transform_kwargs = {k: v for k, v in case_param_dict["transform_kwargs"].items()}
+    if "target_data" in transform_kwargs.keys():
+        if transform_kwargs["target_data"] is not None:
+            transform_kwargs["target_data"] = input[transform_kwargs["target_data"]]
+
+    return (
+        input,
+        {k: v for k, v in case_param_dict["grid_kwargs"].items()},
+        target,
+        transform_kwargs,
+        expected,
+    )
+
+
+# TODO: I am not sure how we would handle periodic axes atm. Should we raise an error?
+
+
+@pytest.fixture(
+    scope="module",
+    params=list(cases().keys()),
+)
+def all_cases(request):
+    return constructor(cases()[request.param])
+
+
+@pytest.fixture(
+    scope="module",
+    params=[c for c in list(cases().keys()) if "linear" in c],
+)
+def linear_cases(request):
+    return constructor(cases()[request.param])
+
+
+@pytest.fixture(
+    scope="module",
+    params=[c for c in list(cases().keys()) if "conservative" in c],
+)
+def conservative_cases(request):
+    return constructor(cases()[request.param])
+
+
+@pytest.fixture(
+    scope="module",
+    params=[c for c in list(fail_cases().keys()) if "conservative" in c],
+)
+def failing_cases(request):
+    return constructor(fail_cases()[request.param])
+
+
+"""Test suite."""
+
+
+def _parse_dim(da):
+    # utility that check that the input array has only one dim and return that
+    assert len(da.dims) == 1
+    return list(da.dims)[0]
+
+
+def test_linear_interpolation_target_value_error():
+    """Test that linear_interpolation throws an error when `target` is a np array"""
+    input, grid_kwargs, target, transform_kwargs, expected = constructor(
+        cases()["linear_depth_depth"]
+    )
+
+    # method keyword is only for high level tests
+    transform_kwargs.pop("method")
+
+    with pytest.raises(ValueError):
+        interpolated = linear_interpolation(input.data, input.z, target.data, "z", "z")
+    # TODO: test with the other method
+
+
+def test_low_level_linear(linear_cases):
+    """Test the linear interpolations on the xarray wrapper level"""
+    input, grid_kwargs, target, transform_kwargs, expected = linear_cases
+
+    # method keyword is only for high level tests
+    transform_kwargs.pop("method")
+
+    input_dim = _parse_dim(input.data)
+    target_dim = _parse_dim(target)
+
+    # parse the target_data manually
+    target_data = transform_kwargs.pop("target_data", None)
+    if target_data is None:
+        target_data = input[input_dim]
+
+    interpolated = linear_interpolation(
+        input.data,
+        target_data,
+        target,
+        input_dim,
+        input_dim,
+        target_dim,
+        **transform_kwargs
+    )
+    xr.testing.assert_allclose(interpolated, expected.data)
```

Comment:
what type of object is `expected`?

Reply:
xr.Dataset

Reply:
Then why not ` xr.testing.assert_allclose(interpolated, expected)`?

Reply:
Because interpolated is just a dataarray at this point. I hope that in the future we can pass a full dataset (thats why I set `expected` up to be a full dataset; its slightly clunkier now but more flexible for the future)